Add VOF Van Keken benchmark section (#3425)
geodynamics/aspect: doc/manual/manual.tex

Setting 
{\tt   set Initial adaptive refinement        = 4}
yields a series of meshes as shown in Fig. (\ref{fig:meshes}), all produced during the 

> first time step. As expected, we see that the location of the highest mesh refinement

Author: ```suggestion
first timestep. As expected, we see that the location of the highest mesh refinement
```

\begin{figure}
   \centering

>    \includegraphics[width=0.7\textwidth]{cookbooks/crustal_deformation/grids.png}

Author: I think some other commits may have been mixed up during the rebase. I will look into correcting that.

\textit{This section is a co-production of Cedric Thieulot, Juliane Dannberg,
Timo Heister and Wolfgang Bangerth with an extension to this benchmark provided by the Virginia Tech Department of Geosciences class ``Geodynamics and ASPECT'' co-taught by Scott King and D.~Sarah Stamps.}

\paragraph{Computation of the van Keken Problem with compositional fields}

> \label{paragraph:van-keken compositional fields}

Reviewer: Just drop the `\paragraph` and use the label from the subsubsection instead (see also my comment below about the new paragraph you add).

\begin{figure}
   \centering
   \includegraphics[width=0.7\textwidth]{cookbooks/crustal_deformation/grids.png}

>    \caption{\it Mesh evolution during the first time step (refinement is based on strain rate).}

Author: I think it was likely an extra fixup that got caught up in the rest (in either direction). I will likely remove it so the diff is cleaner

We can observe that when the two-layer viscosity model is added, there is only one apparent peak for each case. The first peaks of the 0.01 viscosity contrast tests appear earlier and are larger in magnitude than those of 0.1 viscosity contrast tests.  There are no secondary plumes and the whole system tends to reach stability after around 500 time steps.

> \paragraph{Computation of the van Keken Problem with the Volume-of-Fluid Interface Tracking Method}

Reviewer: I think that you should just make this a `\subsubsection` like the other benchmarks/cookbooks. You should also put the note at the top that documents who contributed the section (i.e., *you*).

Reviewer 2: @bangerth This was a collaboration between @class4kayaker and me.  He wrote the first draft and made the computations.  I wrote the final draft.

Reviewer: @egpuckett -- just add something like that as the first sentence of the subsubsection, like is done for most of the other subsubsections.

it consists of two distinct, immiscible fluids and interface tracking algorithms are 
specifically designed not to allow the two fluids to mix.
In particular, assuming the computation is sufficiently well-resolved, the fluids will not mix 
at sub-grid scales over the entire duration of the computation.
However note that this implies that all computations of the van Keken problem made with the VOF 
method must necessarily be with discontinuous initial conditions.
Finally, since one is often interested in a high resolution image of the shape of the interface 
between the two fluids, one can use the output of the VOF method to examine the interface 
within individual cells or within regions consisting of groups of cells.

Another advantage the VOF method has over modeling moving interface problems with compositional 
fields is that for problems in which the interface occupies a relatively small part of the 
computational domain, all of the computational work in the VOF method is done in cells that lie 
in a neighborhood of the interface, rather than in the entire computational domain. 

\begin{figure}[b!]
   \centering

> \includegraphics[width=0.5\textwidth]{cookbooks/benchmarks/vof-van-keken/rms_vel_comparison.png}

Reviewer 3: To my eye, I can really only see the lines for refinements 5-7. Perhaps this is all that is needed to show convergence?

Author: I would appreciate it @egpuckett would comment. It is true that after 7, the lines are overlapping to visual examination on the plot in question, but not including the further lines would make including that point more questionable.

Reviewer 2: Please keep all lines / levels of refinement in.  Unless I missed something we are duplicating exactly what was done in 

> 5.4.3 The van Keken thermochemical composition benchmark

> This section is a co-production of Cedric Thieulot, Juliane Dannberg, Timo Heister and Wolfgang Bangerth with an extension to this benchmark provided by the Virginia Tech Department of Geosciences class “Geodynamics and ASPECT" co-taught by Scott King and D. Sarah Stamps.

We want to compare Apples to Apples.  We think VOF is a much better methodology **for certain problems**, but there seem to be a lot of, or maybe only some influential, ASPECT users and computational geodynamicists that think otherwise and would just as soon this methodology - interface tracking - went away.  Personally, I think that is so crazy that I can't find words to express how backward thinking this is.  Interface tracking is the right way to do certain problems in every field of mechanics, fluid mechanics, not to mention a gazillion other examples.  We need the support of the ASPECT community to implement a 3D VOF method and one that works on any kind of arbitrary grid that is supported by deal.ii.

I think the way to do that is to show everyone how much better the VOF results are **on some problems**, and this one in particular.

   \caption{Evolution of the root mean square velocity as a function of time for computations 
    of the Van Keken problem made with the VOF interface tracking algorithm with five 
    different global mesh refinements.

>     Since the VOF initial conditions are discontinuous, the above results should really be 

Reviewer 3: Perhaps no need to start this sentence on a new line?

Author: I had been adding linebreaks at sentence ends in my new additions relatively frequently due to the fact that latex treats that type of linebreak as standard whitespace with the intent of making it easier on me to look at the diffs.

Reviewer 3: Gotcha. Before merging, please reorganize the text into standard paragraphs and line paragraphs.

Author: My understanding and testing indicates that LaTeX does not insert a paragraph break unless there is a blank line, os the current approach does not actually produce malformed output (as I indicated it combines the sentences into a paragraph as usual), which was part of the reason I used the approach in question. I can do the reformatting as a large block commit as the final change before merging though.

    compared with the computations with discontinuous initial conditions on the left in 
    Figure~\ref{fig:vk-2}, although the above results also compare extremely favorably with the 
    computations with smoothed, continuous initial conditions for the compositional field on 
    the right in Figure~\ref{fig:vk-2}.
    As in Figure~\ref{fig:vk-2}, 5 global refinements correspond to a $32 \times 32$ mesh and 10 global refinements correspond to a $1024 \times 1024$ mesh.}
    \label{fig:vof-vk-1}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.4\textwidth]{cookbooks/benchmarks/vof-van-keken/vof_van_keken_refinement_comparison.png}
    \caption{\it The results of two computations of the Van Keken problem made with the VOF 

>         interface tracking algorithm  overlaid upon each other at $t_{end}=2000$.

Reviewer 3: ```suggestion
        interface tracking algorithm overlaid upon each other at $t_{end}=2000$.
```

        This visualization consists of the \underline{two} zero contours of 

>         $\tilde{\psi}^7 \, (t_{end})$ and $\tilde{\psi}^8 \, (t_{end})$, where 

Reviewer 3: What exactly does `\tilde{\psi}^7` and `\tilde{\psi}^8` refer to? Reading below, my understanding is that `psi` is the zero contour of the reconstructed interface. Do `7` and `8` refer to the refinement level?

Author: Yes, that is correct. I believe @egpuckett modified this to the current phrasing, while my first draft referred to the reconstructed interface, and I would therefore appreciate any comments on whether reverting to the earlier text would make more sense.

Reviewer 3: I would combine the two approaches and perhaps use something like the following wording:
```
This visualization consists of the \underline{two} zero contours of the reconstructed interface for models with refinement levels $\tilde{\psi}^7 \, (t_{end})$ and $\tilde{\psi}^8 \, (t_{end})$, where 
```

Author: I checked and I had made some edits already. I will add my own note on the corresponding region of the relocated text for further discussion before marking this as resolved.

        $\tilde{\psi}^7 \, (t_{end})$ is field where the zero contour is the reconstructed interface 
        at the final time $t_{end}$ computed on a uniform grid with a refinement level of 7 and 

>         similarly for $\tilde{\psi}^8 \, (t_{end})$.

Reviewer 3: This sentence was quite hard to follow. Could you break into two shorter sentences and define `\psi` towards the beginning?

        The color of the zero contour for $\tilde{\psi}^7 \, (t_{end})$ is black and the color of the 
        zero contour $\tilde{\psi}^8 \, (t_{end})$ is bright green.
        Compare with the fourth image on the right in Figure~\ref{fig:vk-1}.
    }
    \label{fig:VOF_van_Keken-02}
\end{figure}

Since, as noted above, when the interface is discontinuous, the van Keken problem is a version 
of the Rayleigh-Taylor problem it is unstable to perturbations of all 
wavelengths,\footnote{This is true whether the two fluids have the same viscosity or different 

> viscosities.} (e.g. see~\cite{SC:1961}), it is therefore, extremely sensitive to the initial 

Reviewer 3: ```suggestion
viscosities.} (e.g. see~\cite{SC:1961}). Therefore, it is extremely sensitive to the initial 
```

conditions.

> Thus, rather than use the standard initialization procedure for the VOF method, which consists 

Reviewer 3: Can you break this down into sentences?

Author: Altered in new location, will mark for comment in my own review before requesting your re-reading.

of integrating the initial interface over each multi-material cell in which one material or 
fluid has the value of $1$ and the other has the value of $0$; i.e., a characteristic function 
representation of the interface~\cite{JMR:2019,JMR-EGP:2019}, instead we compute the initial volume fractions using a signed distance function $\phi$ as follows.

First we create the function $\phi$, which has the following two properties: 1) it is positive 
in the region that contains one of the fluids, which we will refer to as fluid~1, and negative 
in the complement of this region, which we will refer to as fluid~2, and 2) at each point in 

> the domain the magnitude $| \phi |$ of $\phi$  is the distance to the boundary between the 

Reviewer 3: ```suggestion
the domain the magnitude $| \phi |$ of $\phi$ is the distance to the boundary between the 
```

to more accurately reproduce the initial condition on a sub-grid scale than would otherwise be 
possible on the coarser grid on which we compute the time evolution of the interface. 

\vspace{06pt}

\lstinputlisting[language=prmfile]{cookbooks/benchmarks/vof-van-keken/main.part.prm.out}

The relevant sections of the parameter file for this type of initialization of the VOF method 
appears immediately above.
In particular, the combination of \texttt{Number of initialization samples} with the 
\texttt{level set} initialization type indicates that our initialization will consist of 
dividing each grid cell into $16 \times 16$ subcells and the distance to the given initial 
interface $f(x)$, provided in \texttt{Function expression}, is computed in each of the 256 
subcells.
We then use this information to compute a piecewise linear interface approximation to $f(x)$.
The volume fraction in each subcell is then found in the manner described 
in~\cite{JMR:2019,JMR-EGP:2019}.
This initialization procedure provides a much finer and thus, more accurate, initial condition 
than the standard VOF initialization procedure described above.

> % AMR section, currently unused so not included

Reviewer 3: Is the plan to remove the following lines from the cookbook?

Reviewer 4: That means you have a parameter file with AMR and VoF that crashes? Please open an issue about that and post the input file there so that others who encounter this crash will see that it is a known issue.

Reviewer 2: @class4kayaker Is it possible to run this now and see if it works? Given that deal.ii is now v. 9.2.0, perhaps it will work?

% 
% \lstinputlisting[language=prmfile]{cookbooks/benchmarks/vof-van-keken/amr.part.prm.out}

While the visualization configuration in a typical parameter file is sufficient for most 
purposes, when using the VOF method one has the ability to see the division between the fluids 
reconstructed by the VOF algorithm in each cell.
This is accomplished by plotting the zero contour of a field $\tilde\psi$ that
is generated to be $0$ on the reconstructed interface, positive in the region
with fluid~1, and negative in the region with fluid~2.
However $\tilde{\psi}$ does not satisfy the requirement that the magnitude is
equal to the distance to the interface as would be required for the signed
distance function $\tilde{\phi}$.
The modifications to the parameter file that are necessary in order to draw the reconstructed 
boundary as a contour are shown immediately below.
The full configuration file for this version of the benchmark problem can be found at 
\url{cookbooks/van-keken-vof.prm}.

\lstinputlisting[language=prmfile]{cookbooks/benchmarks/vof-van-keken/postprocess.part.prm.out}
\noindent

We made a number of computations of the van Keken problem with the VOF method in order to 

> compare the wall clock times of the VOF results directly with the those computed with a DG

Reviewer 3: ```suggestion
compare the wall clock times with computations using a DG compositional field.
```

> compositional field.

Reviewer 3: Delete if the suggestion above is used.

Author: ```suggestion

```

found in other versions of this benchmark such as the fourth image on the right in 
Figure~\ref{fig:vk-1}.
We also note that the differences in the shape of the interface based on a single refinement 
as shown in Figure~\ref{fig:VOF_van_Keken-02} are minor, although still slightly visible.
This is to be expected as refinement is a perturbation of the initial condition at a smaller 
wave length.

\begin{table}
  \centering
   \begin{tabular}{|c|c|c|c|}
     \hline
      Global Refinement & Number of Processors & VOF & DG \\
      \hline
\phantom{1}5 & \phantom{1}4 &\phantom{6}1.33 minutes\phantom{hours} &\phantom{1}2.57 minutes  \\
\phantom{1}6 & \phantom{1}4 &\phantom{1}8.51 minutes\phantom{hours} & 19.5\phantom{7} minutes \\
\phantom{1}7 & \phantom{1}4 &\phantom{1}1.15 hours\phantom{minutes} & \phantom{1}2.49 hours\phantom{tes} \\
\phantom{1}8 & \phantom{1}4 &\phantom{1}8.53 hours\phantom{minutes} & 19.6 hours\phantom{tes} \\
\phantom{1}9 & 16           &         16.30 hours\phantom{minutes}  & \phantom{1}2.72 days\phantom{tes} \\
          10 & 16           & \phantom{1}5.17 days\phantom{minutess}&>6.00 days\phantom{tess} \\
            \hline
        \end{tabular}

>         \caption{\it Comparison of runtimes for the van Keken problem with VOF and and a DG computational field, in which the initial conditions for DG smoothed as described in section~\ref{paragraph:van-keken compositional fields} above.

Reviewer 3: ```suggestion
        \caption{\it Comparison of runtimes for the van Keken problem with VOF and a DG computational field, in which the initial conditions for DG smoothed are as described in section~\ref{paragraph:van-keken compositional fields} above.
```

Reviewer 2: @class4kayaker @naliboff   I think that the second sentence here:

> ... problem with VOF and and a DG computational field, ...

the word 'computational' is meant to be 'compositional', is it not?

            The times shown are for the full computation, ending at $t_{end} = 2000$ with a CFL
            number of $\sigma=0.5$ in both cases. 
            All of these computations were made with ASPECT version 2.2.0-pre (master, ef542ecc2) in release mode on the Peloton2 cluster at U.C.~Davis.
            We note that the change in the CFL number $\sigma$ and the differing
            choice of cluster makes a direct quantitative comparison between this
            table and table~\ref{tab:runtime-table} invalid due to too many
            confounding factors.
        }
        \label{tab:vof-runtime-comparison-table}
\end{table}

> The consistency of the results shown here differs noticeably from the behavior of the  

Reviewer 3: In this paragraph and some above you are starting multiple sentences on new lines. Is this intentional for the purposes of readability?

Author: Yes, it is intentional, but the primary purpose was to attempt to make diffs cleaner, rather than the primary intent being tied to the readability of the latex file itself.

problem with discontinuous initial conditions when computed with the FEM and DG advection 
algorithms.
One of the possible reasons for these differences is the procedure for 
defining the initial interface.

> As described above, we have designed an approach for setting the initial condition that should 

Reviewer 3: Could you condense the following sentence a bit or break it into two sentences?

reproduce a smooth curve to a high degree of accuracy and, in particular, the curve, which is
where the discontinuity occurs in the discontinuous initial condition version of this problem.

> In order to study this feature of our algorithm, we vary the size of the initial interface 

Reviewer 3: ```suggestion
To study this feature of our algorithm, we vary the size of the initial interface 
```

> perturbation in order to examine the sensitivity of the final results to a small change in the 

Reviewer 3: ```suggestion
perturbation and examine the sensitivity of the final results to a small change in the 
```

initial conditions.
Specifically, we vary the amplitude $a$ of the cosine function in the initial conditions, as 
shown below.

\lstinputlisting[language=prmfile]{cookbooks/benchmarks/vof-van-keken/variation.part.prm.out}

\begin{figure}[htb]
    \centering
    \includegraphics[width=0.5\textwidth]{cookbooks/benchmarks/vof-van-keken/init_diff_rms_vel_comparison.png}
    \caption{\it Computations of the Van Keken problem made with the VOF
        interface tracking algorithm showing the evolution of the RMS velocity
        as a function of time for small changes in the amplitude $a$ of the
        cosine function in the initial condition at 7 levels of refinement.
        Compare to Figures~\ref{fig:vk-6} and~\ref{fig:vof-vk-1}.
    }
    \label{fig:vof-vk-3}
\end{figure}

In these computations we vary the value of $a$ from its usual value of $a = 0.02$ to 
$5\% = 0.001$ below its usual value to $5\%$ above its usual value in increments of $0.01$.

> In other words, we compare the values for $a  =0.019$, $0.020$, and $0.021$.

Reviewer 3: ```suggestion
In other words, we compare the values for $a =0.019$, $0.020$, and $0.021$.
```

Upon examination of Figure~\ref{fig:vof-vk-3}, we see a visible variation in the location of 
the second peak, although the overall shape of the curve remains consistent with the curves 
in Figure~\ref{fig:vof-vk-1}.
The size of this variation in the initial conditions cannot be expected to be reproduced using 
the standard compositional quadrature initialization procedure for VOF unless the cell size is on the scale 
of the change in the value of $a$; i.e., $h \, \lessapprox \, \Delta a = 0.001$.
We also note that the smoothing parameter, which would produce a
$10^{-3}\leq C \leq 1 - 10^{-3}$ band on the order of the same size as the amplitude 

> variation shown here would be approximately $2.8 \cdot 10^{-4}$.

Reviewer 3: ```suggestion
variation shown here, would be approximately $2.8 \cdot 10^{-4}$.
```

This perturbation is much smaller than any of the changes in width of the smoothed regions in 
the computations shown in Figure~\ref{fig:vk-6}.
In summary, these results demonstrate the sensitivity of the discontinuous version of the van 
Keken problem to even extremely small variations in the initial conditions.

\subsubsection{The Rayleigh-Taylor instability}
\label{sec:benchmark-rayleigh-taylor}

\textit{This section was contributed by Cedric Thieulot.}

This benchmark is carried out in \cite{Deu08,Ger10,thie11} and is 

> based on the analytical solution by Ramberg \cite{ramb68}, 

Author: Remove external modification

```suggestion
based on the analytical solution by Ramberg\cite{ramb68}, 
```

which  consists of a gravitationally unstable two-layer system.
Free slip are imposed on the sides while no-slip boundary conditions are imposed on the
top and the bottom of the box.
Fluid 1 $(\rho_1,\eta_1)$ of thickness $h_1$ overlays 
fluid 2 $(\rho_2,\eta_2)$ of thickness $h_2$ (with $h_1+h_2=L_y$).
An initial sinusoidal disturbance of the interface between these

> layers is introduced and is characterized by an amplitude $\Delta$ and a

Author: Remove change outside of new section

```suggestion
layers is introduced and is characterised by an amplitude $\Delta$ and a
```

wavelength $\lambda=L_x/2$ as shown in Figure~\ref{fig:RTi_setup}.

> \begin{figure}[h!]

Author: Left due to possible relevance to correct figure behavior.

Reviewer 4: What do you mean by `possible relevance`? Did you have to add this to create a correct figure, or can it be removed without changes? I would prefer to only add changes that are necessary to resolve issues.

Reviewer: Let's not use `[h!]`. It completely messes up latex's placement algorithm. The default is exactly what we should be using instead. In other words, please revert this change.

        at the final time $t_{end}$ computed on a uniform grid with a refinement level of 7 and 
        similarly for $\tilde{\psi}^8 \, (t_{end})$.
        The color of the zero contour for $\tilde{\psi}^7 \, (t_{end})$ is black and the color of the 
        zero contour $\tilde{\psi}^8 \, (t_{end})$ is bright green.
        Compare with the fourth image on the right in Figure~\ref{fig:vk-1}.
    }
    \label{fig:VOF_van_Keken-02}
\end{figure}

Since, as noted above, when the interface is discontinuous, the van Keken problem is a version 
of the Rayleigh-Taylor problem that is unstable to perturbations of all 
wavelengths,\footnote{This is true whether the two fluids have the same viscosity or different 
viscosities.} (e.g. see~\cite{SC:1961}). Therefore, it is extremely sensitive to the initial 
conditions.
Thus, rather than use the standard initialization procedure for the VOF method, which consists 
of integrating the initial interface over each multi-material cell in which one material or 
fluid has the value of $1$ and the other has the value of $0$; i.e., a characteristic function 

> representation of the interface~\cite{JMR:2019,JMR-EGP:2019}, instead we compute the initial volume fractions using a signed distance function $\phi$ as follows.

Author: Does this suggested edit preserve the necessary information, and make the sentence structure clearer? @egpuckett @naliboff 

```suggestion
In order to address this sensitivity, we do not use a standard composition quadrature initialization procedure
for the VOF method.
Instead we compute the initial volume fractions using a signed distance function $\phi$ as follows~\cite{JMR:2019,JMR-EGP:2019}.
```

Reviewer 3: @class4kayaker - Yes, that rephrasing is much clearer, thanks!

interface $f(x)$, provided in \texttt{Function expression}, is computed in each of the 256 
subcells.
We then use this information to compute a piecewise linear interface approximation to $f(x)$.
The volume fraction in each subcell is then found in the manner described 
in~\cite{JMR:2019,JMR-EGP:2019}.
This initialization procedure provides a much finer and thus, more accurate, initial condition 
than the standard VOF initialization procedure described above.

% AMR section, currently unused so not included
% Due to the change in the advection algorithm, we also modify the refinement
% strategy for the AMR algorithm.
% In particular, the use of the \texttt{volume of fluid interface} refinement
% strategy will ensure that the boundary between that materials, and a small
% region around that will be refined to the maximum extent.\footnote{The
%     inclusion of this refinement strategy is currently required for use of the VOF
%     advection scheme in any model which makes use of AMR due to implementation
% details.}
% The new mesh refinement parameter section is shown below.
% 
% \lstinputlisting[language=prmfile]{cookbooks/benchmarks/vof-van-keken/amr.part.prm.out}

> 

Author: Removal included for easier edit.

```suggestion
```

            The times shown are for the full computation, ending at $t_{end} = 2000$ with a CFL
            number of $\sigma=0.5$ in both cases. 
            All of these computations were made with ASPECT version 2.2.0-pre (master, ef542ecc2) in release mode on the Peloton2 cluster at U.C.~Davis.
            We note that the change in the CFL number $\sigma$ and the differing
            choice of cluster makes a direct quantitative comparison between this
            table and table~\ref{tab:runtime-table} invalid due to too many
            confounding factors.
        }
        \label{tab:vof-runtime-comparison-table}
\end{table}

The consistency of the results shown here differs noticeably from the behavior of the  
problem with discontinuous initial conditions when computed with the FEM and DG advection 
algorithms.
One of the possible reasons for these differences is the procedure for 
defining the initial interface.
As described above, we have designed an approach for setting the initial condition that should 
reproduce a smooth curve to a high degree of accuracy and, in particular, the curve, which is

> where the discontinuity occurs in the discontinuous initial condition version of this problem.

Author: @naliboff, @egpuckett  Would this edit work, or do you have any suggestions on how else to clarify?

```suggestion
As described above, we have designed an approach for setting the initial condition that should
reproduce the smooth curve for the initial boundary between the two fluids for this problem
to a high degree of accuracy.
```

algorithm in ASPECT.
In fact, this problem is particularly well-suited to being computed with the VOF method, since 
it consists of two distinct, immiscible fluids and interface tracking algorithms are 
specifically designed not to allow the two fluids to mix.
In particular, assuming the computation is sufficiently well-resolved, the fluids will not mix 
at sub-grid scales over the entire duration of the computation.
However note that this implies that all computations of the van Keken problem made with the VOF 
method must necessarily be with discontinuous initial conditions.
Finally, since one is often interested in a high resolution image of the shape of the interface 
between the two fluids, one can use the output of the VOF method to examine the interface 
within individual cells or within regions consisting of groups of cells.

Another advantage the VOF method has over modeling moving interface problems with compositional 
fields is that for problems in which the interface occupies a relatively small part of the 
computational domain, all of the computational work in the VOF method is done in cells that lie 

> in a neighborhood of the interface, rather than in the entire computational domain. 

Author: Given there is vastly less work on most of the domain, the implication is not inaccurate, though it is an inherent component of the VOF algorithm, so any implementation that did not deliberately avoid using less computationally intense paths in common cases would also benefit.

> in a neighborhood of the interface, rather than in the entire computational domain. 

Reviewer 2: In other words ... "Yes"

Look at the table Jon made where he compared using a DG compositional field versus VOF on the same problem parameters.

        zero contour $\tilde{\psi}^8 \, (t_{end})$ is bright green.
        Compare with the fourth image on the right in Figure~\ref{fig:vk-1}.
    }
    \label{fig:VOF_van_Keken-02}
\end{figure}

Since, as noted above, when the interface is discontinuous, the van Keken problem is a version 
of the Rayleigh-Taylor problem it is unstable to perturbations of all 
wavelengths,\footnote{This is true whether the two fluids have the same viscosity or different 
viscosities.} (e.g. see~\cite{SC:1961}), it is therefore, extremely sensitive to the initial 
conditions.
Thus, rather than use the standard initialization procedure for the VOF method, which consists 
of integrating the initial interface over each multi-material cell in which one material or 
fluid has the value of $1$ and the other has the value of $0$; i.e., a characteristic function 
representation of the interface~\cite{JMR:2019,JMR-EGP:2019}, instead we compute the initial volume fractions using a signed distance function $\phi$ as follows.

First we create the function $\phi$, which has the following two properties: 1) it is positive 

> in the region that contains one of the fluids, which we will refer to as fluid~1, and negative 

Author: The tilde here is a latex based instruction to insert a space where a line break is not permitted, not a tilde directly.
